### PR TITLE
[test] Port tests in test/ to test/lit/basic/

### PR DESCRIPTION
**Pull request by aheejin**
This ports all tests from `test/` to `test/lit/basic/`. The set of commands and `CHECK` lines used are the same as the ones in #6159. Now we use `lit` to test these, this also deletes all `.wast`, `.wast.from-wast`, `.wast.fromBinary`, and
`.wast.fromBinary.noDebugInfo` files from `test/` and all related test routines from the python scripts.

All `CHECK` lines are generated by `update_lit_checks.py --all-items`.

This also deletes these three multi-memory tests in `test/lit/`, because they seem to contain the same code with the ones in `test/`, which have been ported to `test/lit/basic/` along with other tests.
- `test/lit/multi-memories-atomics64.wast`
- `test/lit/multi-memories-basics.wast`
- `test/lit/multi-memories-simd.wast`

This also adds newlines between `(func`s in case there are none to make `CHECK` lines easy to view, and removes some extra existing newlines here and there.

### CODE REVIEW

**Comment by aheejin** [comment]
- `gc.wast` was deleted in #5858 but not other `.wast.*` files.
- `anyref.wast` was deleted in #2451 but not other `.wast.*` files. Those dangling `.wast.*` files were later renamed in bulk to `externref.wast.*` in #2900.
- I realized I only ported `.wast`. `hello_world.wat` and `unit.wat` have the `.wat` extension haven't been ported. Will port them.
- `tail-call.wast` was moved (and new functions added) to `lit/tail-call.wast` in #4032 but other `.wast.*` files remained.

I also see some dangling tests in `test/wasm2js` which are causing wasm2js test failures. Probably will submit a separate PR to clean up some dangling tests...

**Comment by aheejin** [comment]
I didn't know this, but it looks wasm2js tests use some source files from `test/` and `test/spec/`, and not only from `test/wasm2js/`: https://github.com/WebAssembly/binaryen/blob/1d615b38dd4152494d2f4d3520c8b1d917624a30/scripts/test/wasm2js.py#L51

I guess I just need to redirect them to `test/lit/basic/`...

**Comment by aheejin** [comment]
- Removed test checking/updating routines for `.wast.from-wast`, `.wast.fromBinary`, and `.wast.fromBinary.noDebugInfo`. 

- Ported two `wat` tests (`hello_world.wat` and `unit.wat`) to `test/lit/basic/`, but left the original `hello_world.wat` in place in `test/`, because `wasm_opt.py` does some special tests with it and one of them requires the output to be the same as the input:
https://github.com/WebAssembly/binaryen/blob/1d615b38dd4152494d2f4d3520c8b1d917624a30/scripts/test/wasm_opt.py#L26-L42
https://github.com/WebAssembly/binaryen/blob/1d615b38dd4152494d2f4d3520c8b1d917624a30/scripts/test/wasm_opt.py#L139-L143

**Comment by aheejin** [comment]
The wasm2js failure in the CI will be fixed after #6164 lands.

**Review by kripken — APPROVED** [review]

File: scripts/test/wasm_opt.py [hunk 1]
```diff
@@ -104,35 +104,15 @@ def check():
...
-            shared.binary_format_check(t, wasm_as_args=[], binary_suffix='.fromBinary.noDebugInfo')  # test without debuginfo
-
-        shared.minify_check(t)
-
-    print('\n[ checking wasm-opt debugInfo read-write... ]\n')
-
 
 def update_wasm_opt_tests():
-    print('\n[ checking wasm-opt -o notation... ]\n')
+    print('\n[ updating wasm-opt -o notation... ]\n')
```
Comment (aheejin) [inline scripts/test/wasm_opt.py:109]: Drive-by fix: checking->updating in `update_wasm_opt_tests` messages

File: scripts/test/wasm_opt.py [hunk 2]
```diff
@@ -104,35 +104,15 @@ def check():
...
-        shared.fail_if_not_identical_to_file(actual, f)
-
-        # FIXME Remove this condition after nullref is implemented in V8
-        if 'reference-types.wast' not in t:
-            shared.binary_format_check(t, wasm_as_args=['-g'])  # test with debuginfo
-            shared.binary_format_check(t, wasm_as_args=[], binary_suffix='.fromBinary.noDebugInfo')  # test without debuginfo
-
-        shared.minify_check(t)
-
-    print('\n[ checking wasm-opt debugInfo read-write... ]\n')
```
Comment (aheejin) [inline scripts/test/wasm_opt.py:125]: This line should've been removed in #6162

**Comment by aheejin** [comment]
Any more concerns on landing this PR?

**Review by kripken — APPROVED** [review]

**Review by tlively — APPROVED** [review]
LGTM!
